src/compiler/sys.ts

```diff
@@ -340,11 +346,22 @@ namespace ts {
                     }
 
                     function fileChanged(curr: any, prev: any) {
-                        if (+curr.mtime <= +prev.mtime) {
+                        const isCurrZero = +curr.mtime === 0;
+                        const isPrevZero = +prev.mtime === 0;
+                        const created = !isCurrZero && isPrevZero;
+                        const deleted = isCurrZero && !isPrevZero;
+
+                        const eventKind = created

```

Comment:
I think the existence of the constants makes the code more readable, but I agree that the `if` is better expressed in terms of `eventKind`.